scipy/integrate/_cubature.py

```diff
@@ -8,6 +8,8 @@
 
 import numpy as np
 
+from numpy.polynomial.polynomial import Polynomial
```

Comment:
I think we definitely want to stay away from this! If there's something really needed, we can discuss, but there are a few good reasons to avoid `Polynomial` (for one, it won't go on the GPU).